tests/benchdnn/graph/input_displacer.cpp

```diff
@@ -22,6 +22,58 @@
 
 namespace graph {
 
+namespace {
+
+void handle_special_dt_set(
+        ::graph::deserialized_op_t &op, const ::std::string &dt) {
+    auto driver = opkind2driver(opstr2kind(op.kind_));
+    bool is_f8_quantization = (dt == "f8_e5m2" || dt == "f8_e4m3");
+
+    if (op.in_lts_.size() > 1) {
+        // Matmul/Conv/Deconv have limited support for quantized configurations.
+        if (op.kind_ == "MatMul" || op.kind_ == "Convolution"
+                || op.kind_ == "ConvTranspose") {
+            if (dt == "u8") {
+                // None of them supports u8u8, replace with u8s8.
+                op.in_lts_[1].data_type_ = "s8";
+            } else if (dt == "s4" || dt == "u4") {
+                // None of them supports x4x4, replace with f32x4f32 or
+                // xf16x4xf16.
+                op.in_lts_[0].data_type_ = op.out_lts_[0].data_type_;
+            }
+        }
+    }
+    if (driver == dnnl_driver_t::pool || driver == dnnl_driver_t::binary
+            || is_f8_quantization) {
+        // pool does not support x8f32 on cpu, and binary does not support
+        // x8x8bf16 on gpu, hence replace output with x8.
+        // f8 data types needs setting output data type to f8
+        op.out_lts_[0].data_type_ = dt;
+    } else if (op.out_lts_[0].data_type_ != "bf16") {
+        if (op.in_lts_.size() > 1 && op.in_lts_[1].data_type_ == "s8") {
+            // Use u8 as output data type for two-input operations to avoid
+            // data overflow due to the specific driver logic.
+            op.out_lts_[0].data_type_ = "u8";
+        } else {
+            // Use f32 as output data type since not all primitives support
+            // different data types for input and output.
+            op.out_lts_[0].data_type_ = "f32";
+        }
+    }
+}
+
+int init_ref_prim_and_fill_data(
```

Comment:
The function mixes `SAFE` and `SAFE_V` and no callers check for a status. If it's not intended to recover from the failure and report it through the stack, just construct and return this shared_ptr here. I can't see why ctor shouldn't be a part of this function.

If recovery is intended, SAFE_V should be converted into SAFE, the status must be checked and only a bare object created on a stack should be passed by reference to be constructed and assigned here in case of success.